open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataEventConsumer.java

```diff
@@ -338,20 +334,10 @@ private boolean checkForFullyProcessedMessages() {
         }
         log.info("Checking for fully processed messages whose offsets need to be committed");
 
-        Map<TopicPartition, Queue<KafkaIncomingEvent>> mapToCheck = null;
-        
-        //Make a copy to minimize the amount of time we need to spent
-        //holding a lock on the map.  If we do not synchronize here, a
-        //concurrent modification exception will be thrown if an
-        //attempt is made to update the map while we are iterating though it
-        synchronized(unprocessedEventQueues) {
-            mapToCheck = new HashMap<TopicPartition, Queue<KafkaIncomingEvent>>(unprocessedEventQueues);

```

Comment:
If it's faster , do we still need this mapToCheck HashMap, it would be far more efficient not to actually make a copy.
It would be useful to understand any problem with locking the unprocessedEventQueues

Reply:
That's exactly that we don't need it, I deleted this code. And we also dont need this lock, in principle, when working atomic collections and therefore I removed it.